Pages/Pag81751.NSPPerformanceTest.al

```diff
@@ -152,119 +195,177 @@ page 81751 "NSP Performance Test"
         CreateNumberSeries();
     end;
 
-    local procedure InitEntryTable()
+    local procedure TestNumberIncrement(ShowMessage: Boolean; BulkInsert: Boolean)
     var
-        EntryTable: Record "NSP Entry Table";
-        EntryTable2: Record "NSP Entry Table2";
         StartDateTime: DateTime;
         i: Integer;
+        Message: text[250];
+        StartSqlStatements, StartSqlRowsRead : BigInteger;
     begin
-        EntryTable.DeleteAll();
-        EntryTable2.DeleteAll();
+        DeleteEntries();
+        StartSqlStatements := SessionInformation.SqlStatementsExecuted;
+        StartSqlRowsRead := SessionInformation.SqlRowsRead;
+        if BulkInsert then begin
+            Message := 'Number Inserts with Bulk inserts'
+        end else
+            Message := 'Number Inserts without Bulk inserts';
+
+
         StartDateTime := CurrentDateTime();
 
         for i := 1 to NumberOfIterations do begin
-            EntryTable.Init();
-            EntryTable."Entry No." := i;
-            EntryTable.Description := 'test';
-            EntryTable.Insert(true);
+            CreateEntry(i, BulkInsert);
         end;
+        IsEntryTableReady := true;
+        Result.Log(Message, CurrentDateTime() - StartDateTime, NumberOfIterations, comment, SessionInformation.SqlRowsRead - StartSqlRowsRead, SessionInformation.SqlStatementsExecuted - StartSqlStatements);
 
-        Message('Tables Initialization: %1 records took %2.', NumberOfIterations, CurrentDateTime() - StartDateTime);
+        Commit;
+        if ShowMessage then
+            Message('Tables Initialization: %1 records took %2.', NumberOfIterations, CurrentDateTime() - StartDateTime);
```

Comment:
@NAVFreak 
In this case duration (CurrentDateTime() - StartDateTime) will always be greater than in ResultLog

Reply:
Yeah, you are correct. I wouldn't do this in prod apps and I only was interested in the RunAllTests.